dspace-api/src/main/java/org/dspace/app/itemupdate/AddBitstreamsAction.java

```diff
@@ -210,7 +210,7 @@ protected String addBitstream(Context context, ItemArchive itarch, Item item, Fi
 				if (group != null)
 				{
                     authorizeService.removeAllPolicies(context, bs);  // remove the default policy
-                    authorizeService.createResourcePolicy(context, bs, group, null, ce.permissionsActionId, null);
+                    authorizeService.createResourcePolicy(context, bs, group, null, ce.permissionsActionId, null, null, null, null, null);
```

Comment:
I'm still seeing a large number of these `createResourcePolicy()` calls that now end in 5 `null` parameters.  While it's not necessarily bad code, it looks very odd / unclear to me (and is extremely duplicative).  So, as I mentioned in #1889, I wonder if we should have multiple `createResourcePolicy()` methods, with different sets of attributes passed in, rather than continually passing in nulls for the majority of the params.

So, for example, retain the existing `authorizeService.createResourcePolicy(context, bs, group, null, ce.permissionsActionId, null)` call in this line, and have its method code now call another `createResourcePolicy()` method with all 10 params, and the last 4 being set to `null`.  This would cut down on a number of changes in the PR which simply append 4 more `null` attributes on every `createResourcePolicy` call.

Does this make sense? Or am I missing something in why we'd rather do it this way?

Reply:
Yes, I agree. I left the method `createResourcePolicy()` after refactoring the method to highlight the changes, no other reason.